Add importance metric

VioPaige opened the pull request:
Hello, recently I noticed the popularity metric in qBitTorrent, and when talking with a friend, we realised that the original idea of stopping seeding simply because a torrent is not that popular is fairly shortsighted when there are only very few seeds, my hope is that this metric can help against users potentially removing rarely seeded torrents.

## Importance
Importance is a useful metric for tracking which torrents are more important to keep and which can be 'safely' discarded, as it is not helpful to delete unpopular torrents if there is only a handful of seeds.
Importance is calculated as `Popularity / Total Seeds`, such that the importance soars when the popularity is relatively high and the seed count is relatively low.

## Appearance
![image](https://github.com/user-attachments/assets/ed0d674e-44da-4e63-8eb2-46a6eb78a31e)

I hope other users appreciate this feature as much as I will.

Kind Regards,
VioPaige

dyseg commented:
I find this feature interesting, I gave it a try. I would like to suggest some changes/fixes for the WebUI:
* change infinite value to -1 during serialization, otherwise it is passed as null
* display infinite symbol in torrent list and in properties window
* display the importance metric in the same line to match the GUI's layout
```diff
diff --git a/src/webui/api/serialize/serialize_torrent.cpp b/src/webui/api/serialize/serialize_torrent.cpp
index 9a292c996..4f67ad5d4 100644
--- a/src/webui/api/serialize/serialize_torrent.cpp
+++ b/src/webui/api/serialize/serialize_torrent.cpp
@@ -169,7 +169,7 @@ QVariantMap serialize(const BitTorrent::Torrent &torrent)
         {KEY_TORRENT_RATIO, adjustRatio(torrent.realRatio())},
         {KEY_TORRENT_RATIO_LIMIT, torrent.ratioLimit()},
         {KEY_TORRENT_POPULARITY, torrent.popularity()},
-        {KEY_TORRENT_IMPORTANCE, torrent.importance()},
+        {KEY_TORRENT_IMPORTANCE, adjustRatio(torrent.importance())},
         {KEY_TORRENT_SEEDING_TIME_LIMIT, torrent.seedingTimeLimit()},
         {KEY_TORRENT_INACTIVE_SEEDING_TIME_LIMIT, torrent.inactiveSeedingTimeLimit()},
         {KEY_TORRENT_LAST_SEEN_COMPLETE_TIME, Utils::DateTime::toSecsSinceEpoch(torrent.lastSeenComplete())},
diff --git a/src/webui/www/private/scripts/dynamicTable.js b/src/webui/www/private/scripts/dynamicTable.js
index 0ac12cd3c..3976a4c4b 100644
--- a/src/webui/www/private/scripts/dynamicTable.js
+++ b/src/webui/www/private/scripts/dynamicTable.js
@@ -1437,7 +1437,7 @@ window.qBittorrent.DynamicTable ??= (() => {
             };
 
             // importance
-            this.columns["importance".updateId] = function(td, row) {
+            this.columns["importance"].updateTd = function(td, row) {
                 const value = this.getRowValue(row);
                 const importance = (value === -1) ? "∞" : window.qBittorrent.Misc.toFixedPointString(value, 2);
                 td.textContent = importance;
diff --git a/src/webui/www/private/scripts/prop-general.js b/src/webui/www/private/scripts/prop-general.js
index 926b9900c..8fa0095cc 100644
--- a/src/webui/www/private/scripts/prop-general.js
+++ b/src/webui/www/private/scripts/prop-general.js
@@ -174,8 +174,10 @@ window.qBittorrent.PropGeneral ??= (() => {
                     document.getElementById("share_ratio").textContent = data.share_ratio.toFixed(2);
 
                     document.getElementById("popularity").textContent = data.popularity.toFixed(2);

-                    document.getElementById("importance").textContent = data.importance.toFixed(2);
+                    const importance = (data.importance === -1)
+                        ? "∞"
+                        : data.importance.toFixed(2);
+                    document.getElementById("importance").textContent = importance;
 
                     document.getElementById("reannounce").textContent = window.qBittorrent.Misc.friendlyDuration(data.reannounce);
 
diff --git a/src/webui/www/private/views/properties.html b/src/webui/www/private/views/properties.html
index 8b8beccd1..17e947ef4 100644
--- a/src/webui/www/private/views/properties.html
+++ b/src/webui/www/private/views/properties.html
@@ -52,8 +52,6 @@
                 <tr>
                     <td class="generalLabel" title="QBT_TR(Ratio / Time Active (in months), indicates how popular the torrent is)QBT_TR[CONTEXT=PropertiesWidget]">QBT_TR(Popularity:)QBT_TR[CONTEXT=PropertiesWidget]</td>
                     <td id="popularity" title="QBT_TR(Ratio / Time Active (in months), indicates how popular the torrent is)QBT_TR[CONTEXT=PropertiesWidget]"></td>
-                </tr>
-                <tr>
                     <td class="generalLabel" title="QBT_TR(Popularity / Total Seeds (Higher number implies it is more 'important' to seed the torrent))QBT_TR[CONTEXT=PropertiesWidget]">QBT_TR(Importance:)QBT_TR[CONTEXT=PropertiesWidget]</td>
                     <td id="importance" title="QBT_TR(Popularity / Total Seeds (Higher number implies it is more 'important' to seed the torrent))QBT_TR[CONTEXT=PropertiesWidget]"></td>
                 </tr>

```
Before:

![image](https://github.com/user-attachments/assets/f5d53069-8935-4b45-b356-69d80465ab0b)

After:

![image](https://github.com/user-attachments/assets/0241936d-f6e2-4f40-ba6b-c30c5b76fa8e)

VioPaige commented:
@dyseg I have implemented your suggested changes, but have not tested yet (I also modified the location in the normal ui), I will when I get on my linux device later today.

VioPaige commented:
I have checked it now and everything seems correct

xavier2k6 commented:
@qbittorrent/bug-handlers @stalkerok @HanabishiRecca thoughts?

VioPaige commented:
While the metric might be unreliable I think the relative amounts (compared between torrents being uploaded from the same device) should still tell a story, based on which removing less useful torrents can be done (which is useful for me personally, perhaps not many other people are not very interested, but I do know about some people that are).
